### PR TITLE
AY-7222 Fix otio_review no handles and tempdir for Resolve

### PR DESCRIPTION
**Pull request by robin-ynput**
## Changelog Description

Resolve https://github.com/ynput/ayon-resolve/issues/44

* Ensure `ayon-core.pipeline.tempdir.create_custom_tempdir` is still exposed but flagged as backward compatible.
  This function is not officially exposed in the `__init__.py` but directly called from within resolve:
  https://github.com/search?q=org%3Aynput+create_custom_tempdir&type=code
  
* Ensure extract OTIO review does not expect handle values when no otioClip is set

## Testing notes:
Testing procedure through Resolve is here: https://github.com/ynput/ayon-resolve/pull/46

### CODE REVIEW

**Review by iLLiCiTiT — APPROVED** [review]
Small optional comment, otherwise looks good.

**Review by iLLiCiTiT — CHANGES_REQUESTED** [review]
Noticed one incomatibility.

File: client/ayon_core/pipeline/tempdir.py [hunk 1]
```diff
@@ -70,6 +71,18 @@ def _create_local_staging_dir(prefix, suffix, dirpath=None):
     )
 
 
+def create_custom_tempdir(project_name, anatomy):
```
[the remark this answers is not in the document]
Comment (robin-ynput) [reply client/ayon_core/pipeline/tempdir.py]: You're right, this is adjusted. Thanks for this !

**Comment by robin-ynput** [comment]
> Noticed one incomatibility.

This is adjusted ! Cheers.

**Pull request merged**